frontend/src/router.js

```diff
@@ -12,86 +12,107 @@ export default new Router({
+        default: () => import(/* webpackChunkName: "about" */ './views/Home'),
+        bottombar: () => import(/* webpackChunkName: "navigation" */ './views/BottomBar')
+      },
+      beforeEnter: requireAuth
+    },
+    {
+      path: '/profile',
+      name: 'profile',
+      components: {
+        topbar: () => import(/* webpackChunkName: "navigation" */ './views/TopBar'),
+        default: () => import(/* webpackChunkName: "about" */ './views/Profile'),
+        bottombar: () => import(/* webpackChunkName: "navigation" */ './views/BottomBar')
+      },
       beforeEnter: requireAuth
     },
     {
       path: '/camps',
       name: 'camps',
       components: {
-        default: () => import(/* webpackChunkName: "camps" */ './views/Camps.vue')
+        topbar: () => import(/* webpackChunkName: "navigation" */ './views/TopBar'),
+        default: () => import(/* webpackChunkName: "camps" */ './views/Camps'),
+        bottombar: () => import(/* webpackChunkName: "navigation" */ './views/BottomBar')
       },
       beforeEnter: requireAuth
     },
     {
       path: '/camps/:campId/:campTitle?',
       components: {
-        default: () => import(/* webpackChunkName: "camp" */ './views/Camp.vue'),
-        aside: () => import(/* webpackChunkName: "camps" */ './views/Camps.vue')
+        topbar: () => import(/* webpackChunkName: "navigation" */ './views/camp/TopBar'),
+        default: () => import(/* webpackChunkName: "camp" */ './views/camp/Camp'),
+        aside: () => import(/* webpackChunkName: "periods" */ './views/camp/Periods'),
+        bottombar: () => import(/* webpackChunkName: "navigation" */ './views/camp/BottomBar')
       },
       beforeEnter: requireAuth,
       props: {
-        default: route => ({ camp: campFromRoute(route) })
+        default: route => ({ camp: campFromRoute(route) }),
+        aside: route => ({ camp: campFromRoute(route) })
       },
       children: [
         {
           path: 'collaborators',
           name: 'camp/collaborators',
-          component: () => import(/* webpackChunkName: "campCollaborators" */ './components/camp/Collaborators.vue'),
-          meta: { layout: 'camp' }
-        },
-        {
-          path: 'periods',
-          name: 'camp/periods',
-          component: () => import(/* webpackChunkName: "campPeriods" */ './components/camp/Periods.vue'),
-          meta: { layout: 'camp' }
+          component: () => import(/* webpackChunkName: "campCollaborators" */ './views/camp/Collaborators.vue')
         },
         {
-          path: 'picasso',
-          name: 'camp/picasso',
-          component: () => import(/* webpackChunkName: "campPicasso" */ './components/camp/CampPicasso.vue'),
-          meta: { layout: 'camp' }
+          path: 'overview',
+          name: 'camp/overview',
+          component: () => import(/* webpackChunkName: "campPicasso" */ './views/camp/CampOverview'),
+          beforeEnter: parseBooleanInQuery
```

Comment:
Ist das wirklich nötig? Könntest du nicht in CampOverview.vue stattdessen `this.$route.query.list === 'true'` überprüfen?

Reply:
Das hat leider nicht funktioniert. Das hat nicht geklappt, wenn man mit ?list=false eingestiegen ist.

Reply:
Okay, I see. Den False state könnten wir aber durch Abwesenheit des Query Params anzeigen. So gehts ohne parseBooleanInQuery:
```
:to="{ query: { ...this.$route.query, list: !this.listFormat || undefined } }">
```
Das `|| undefined` bewirkt, dass der Query Parameter ganz entfernt wird wenn er false sein sollte. Dann kommt man auch mit dem Check `this.$route.query.list` aus (ohne `=== 'true'`).

Reply:
`list: !this.listFormat` ist dann eigentlich ein `!(this.listFormat == 'true')` und nicht `===`

Reply:
Wenn man jetzt in der URL ?list=false eingibt wird trotzdem eine Liste angezeigt. Ist das uns egal?

Reply:
Wenn du die `''` um `true` aus deiner Aussage rausnimmst stimme ich dir zu.

Reply:
Einziger Nachteil: Wenn der User von Liste auf Wochenansicht umschaltet bleibt im `query` objekt der Key `list: undefined`. Das ist an sich kein Problem, ausser beim `exact` route matching welches wir fürs Highlighting des `v-btn` in der TopBar verwenden. Konkret: Wenn du auf den Button drückst ist der v-btn in der TopBar nicht mehr highlighted. Aber das ist bereits jetzt so wenn man in der Listenansicht ist (weil der Query Param nicht exact matched).

Als Lösung würde ich entweder das Highlighting für diesen Button ganz ausschalten (wie auch immer, Vuetify macht das automatisch rein bei `:to`), oder von Hand die active class hinzufügen: `:class="{ 'v-btn--active': someComputedConditionThatIgnoresTheQueryParams }"`
Siehe auch: https://forum.vuejs.org/t/how-to-remove-one-query-string-from-url/39176/4 und https://github.com/vuejs/vue-router/issues/2040

Reply:
Habe das Problem gelöst, indem ich einen Redirect von `path: ''` auf `camp/overview` mache und beim Button `exact` entfernt habe. So ist `/overview` standardmässig drin. Finde ich verständlicher, als mit dem `alias`.

Reply:
> Wenn man jetzt in der URL ?list=false eingibt wird trotzdem eine Liste angezeigt. Ist das uns egal?

Ja das wäre mir egal, denn die URLs werden ja eigentlich immer von Vue korrekt generiert. Ich finde wir müssen es nicht unterstützen dass der User list=false tippen kann wenn er den list-Parameter genausogut weglöschen kann.

Reply:
> Habe das Problem gelöst, indem ich einen Redirect von `path: ''` auf `camp/overview` mache und beim Button `exact` entfernt habe. So ist `/overview` standardmässig drin. Finde ich verständlicher, als mit dem `alias`.

Ich sehe im Code aber aktuell kein `path: ''` und auch keinen Redirect. Wie ist es denn jetzt wirklich?

Und wenn du `exact` entfernst ist der Button eben auch aktiv wenn man in /admin oder /collaborators ist, ist dir das bewusst?